### ISSUE TITLE
repo contains 107 files with windows CRLF line endings

### ISSUE THREAD

**Issue by jspillers** [finos/perspective#1024]
## Bug Report

There are 107 files in the repo that contain CRLF line endings as opposed to LF. 

### Steps to Reproduce:
 1. `cd perspective`
 2. `find . | grep -v dist | grep -v node_modules | xargs file | grep 'CRLF' | sed 's/:.*$//'`

### Expected Result:
above one-liner should return zero lines

### Actual Result:
it returns 107 files with windows line endings

**Comment by MisterAwesome23** [comment]
Hello. I am new to perspective and would love to contribute.
I would like to fix this issue, if no one is already on it.

> windows line endings

If I got this right- there are 107 files ending with "\r\n" instead of a "\n"
More on this [here](http://www.cs.toronto.edu/~krueger/csc209h/tut/line-endings.html)
So we need files ending in the unix format of a "\n" right?

**Comment by timkpaine** [comment]
@MisterAwesome23 correct

**Comment by MisterAwesome23** [comment]
I've replaced all the ending by unix endings now. 

> ```find . | grep -v dist | grep -v node_modules | xargs file | grep 'CRLF' | sed 's/:.*$//'```

This now results in 0 files.

Link to entire changes-
https://pastebin.com/A7Pjn3wB

Creating a pull for this.

**Comment by texodus** [comment]
Thanks for the report!

We've gotten a few "ctrl-f" style fixes for this, so it's worth clarifying what we would consider a proper fix for this issue.  It's not quite enough to simply remove the CRLF characters - we don't want future ones committed either.

Our linter `ESLint` supports [`linebreak-style`](https://eslint.org/docs/rules/linebreak-style) and the documentation indicates this should be set to `\n` by default.  It is clearly not working - but we should find out why and fix it, then this will be asserted as a git hook via `yarn lint`, preventing future commits from containing bad characters, and the existing errors can be trivially repaired with `yarn fix` (which will prevent any PRs for this meeting copious merge conflicts).

**Comment by darjama** [comment]
I've submitted a pull request  #1075 to modify the eslint settings and change the .gitattributes file so that .js files aren't altered to have CRLF line endings on checkout on a windows machine as documented [here](https://eslint.org/docs/rules/linebreak-style#using-this-rule-with-version-control-systems). 

The pull request doesn't change the existing files, and at least on my machine using `eslint --stdin --fix` brings my processor to its knees with so many changes. It works if I go directory by directory though. [This tool](https://marketplace.visualstudio.com/items?itemName=vs-publisher-1448185.keyoti-changeallendoflinesequence#:~:text=Open%20the%20command%20pallette%20(CTRL,changed%20(directories%20starting%20with%20'.) for VSCode is helpful for batching the changes for all files.
I've submitted an ICLA.
